tool.ts

```diff
@@ -222,7 +222,7 @@ export async function downloadTool(url: string, fileName?: string): Promise<stri
             // TODO: retries
             tl.debug('downloading');
             let response: httpm.HttpClientResponse = await http.get(url);
-            if (response.message.statusCode != 200) {
+            if (response.message.statusCode >= 400) {
```

Comment:
We don't need this if typed rest client is handling 3xx.  if there's an large / infinite redirect issue, typed rest client will appropriately error and this shouldn't mask it.  Same for 500s.